hvac/v1/__init__.py

```diff
@@ -18,6 +21,48 @@
     has_hcl_parser = False
 
 
+# TODO(v4.0.0): remove _sentinel and _smart_pop when write no longer has deprecated behavior:
+# https://github.com/hvac/hvac/issues/1034
+_sentinel = object()
+
+
+def _smart_pop(
+    dict: dict,
+    member: str,
+    default: t.Any = _sentinel,
+    *,
+    posvalue: t.Any = _sentinel,
+    method: str = "write",
+    replacement_method: str = "write_data",
+):
+    try:
+        value = dict.pop(member)
+    except KeyError:
+        if posvalue is not _sentinel:
+            return posvalue
```

Comment:
Maybe its better to have only one return (outside of else at the end) and do `value = posvalue` and for the next elif `value = default` for readability ?

Reply:
Maybe personal preference but I prefer the `return` for readability, otherwise I have to follow the rest of the flow to see where the value gets used again and/or returned.